source/FreeRTOS_IP_Timers.c

```diff
@@ -382,17 +382,19 @@ static BaseType_t prvIPTimerCheck( IPTimer_t * pxTimer )
 /**
  * @brief Enable/disable the TCP timer.
  *
- * @param[in] xEnableState: pdTRUE - enable timer; pdFALSE - disable timer.
+ * @param[in] xExpiredState: pdTRUE - set as expired; pdFALSE - set as non-expired.
  */
-    void vIPSetTCPTimerEnableState( BaseType_t xEnableState )
+    void vIPSetTCPTimerExpiredState( BaseType_t xExpiredState )
     {
-        if( xEnableState != pdFALSE )
+        xTCPTimer.bActive = pdTRUE_UNSIGNED;
```

Comment:
If the timer is just being marked as expired, why do we need to mark it as active?
Is there any place where we will mark it as inactive thus potentially requiring the timer to be set as active again?

Reply:
@AniruddhaKanhere wrote:
> If the timer is just being marked as expired, why do we need to mark it as active?

Because when a timer is inactive, it will be ignored by the function `prvIPTimerCheck()`:
~~~c
static BaseType_t prvIPTimerCheck( IPTimer_t * pxTimer )
{
    BaseType_t xReturn;

    if( pxTimer->bActive == pdFALSE_UNSIGNED )
    {
        /* The timer is not enabled. */
        xReturn = pdFALSE;
    }
    else
~~~

and so `prvIPTimerCheck()` will never see the expired flag :-(

> Is there any place where we will mark it as inactive thus potentially requiring the timer to be set as active again?

The flag `bActive` is set during every call to `vIPSetTCPTimerExpiredState()`, while nobody will ever clear it. It would be enough to set it once at start-up.

At the other hand, that would require a new function `vIPSetTCPTimerEnableState()` and an extra function-call at start-up, and one more unit test... without a measurable performance gain.